hack/tools/release/notes/process.go

```diff
@@ -132,8 +132,13 @@ func (g prEntriesProcessor) generateNoteEntry(p *pr) *notesEntry {
 		// Release trigger PRs from previous releases are not included in the release notes
 		return nil
 	case strings.HasPrefix(entry.title, ":seedling:"), strings.HasPrefix(entry.title, "🌱"):
+		title := removePrefixes(entry.title, []string{":seedling:", "🌱"})
+		// Skip PRs bump dependencies as they will be listed separately.
+		if area == "Dependency" {
```

Comment:
We are flagging a lot more than just dependabot go dependency bumps with area/dependency (e.g. Go, Trivy, ... bumps, but I think even more).

I think dropping all area/dependency PRs is too much, we might lose something important there.

Can we drop all PRs from dependabot instead? (we would additionally drop GitHub actions bumps, but I think that is fine)

Reply:
(also works better with other repositories that don't set the label but are also using dependabot like CAPV and controller-runtime)

Reply:
I'm trying to filter the GitHub API response for the author but I'm a bit confused on how the code works here. How do we go from returning the `collection` [here](https://github.com/kubernetes-sigs/cluster-api/blob/main/hack/tools/release/notes/github.go#L204) to the `prs []githubPR` object [here](https://github.com/kubernetes-sigs/cluster-api/blob/main/hack/tools/release/notes/github.go#L152)? I tried adding an `author` field to the `githubPR` type, but I'm a bit confused where the existing fields (Number, Title, Labels) are populated. `pageRead` just seems to give me the entire PR page. @g-gaston do you know if there's a way to add an author field to the current flow?

Reply:
Sorry I don't fully understand the question. If the search api returns an author field for pr objects (I'm not sure if it does, you would need to check the docs or just run a query yourself and check the output), then adding it to the `githubPR` struct will make that available when calling `listMergedPRs`.

If you want to make it available to the processor (and the printer), you'll need to add the field to the `pr` struct as well: https://github.com/kubernetes-sigs/cluster-api/blob/main/hack/tools/release/notes/list.go#L117-L121

What I would do here is add the author information to the `pr` domain object so you have it available in the processor (this is the one responsible from processing each pr and deciding which kind of note entry it should be). In your case it seems you want the processor to discard PRs based on the author, so they should no be converted to notes entries. Beware this will affect the total number of commits printed at the beginning of the notes. If you don't want this to happen, then maybe slightly changing the responsibilities of processor and printer is a better idea.

Reply:
> Can we drop all PRs from dependabot instead (filter on author:app/dependabot )?

There still seems to be a good amount of dependencies bump PRs that are not created by dependabot. Should we just leave those as is?

Reply:
I think it's better if we sort through manual bump PRs manually compared to dropping all dependency PRs from the release notes.

We even might just leave them in there. Usually we only create manual bumps if either dependabot can't bump something or if we have to bump manually because we have to change more.


Maybe let's just go with dropping PRs from author dependabot and leaving everything else in the release notes.